src/linear_algebra/tensor_product.lean

```diff
@@ -910,16 +922,7 @@ lemma tmul_sub (m : M) (n₁ n₂ : N) : m ⊗ₜ (n₁ - n₂) = (m ⊗ₜ[R] n
 lemma sub_tmul (m₁ m₂ : M) (n : N) : (m₁ - m₂) ⊗ₜ n = (m₁ ⊗ₜ[R] n) - (m₂ ⊗ₜ[R] n) :=
 (mk R M N).map_sub₂ _ _ _
 
-/--
-While the tensor product will automatically inherit a ℤ-module structure from
-`add_comm_group.int_module`, that structure won't be compatible with lemmas like `tmul_smul` unless
-we use a `ℤ-module` instance provided by `tensor_product.semimodule'`.
-
-When `R` is a `ring` we get the required `tensor_product.compatible_smul` instance through
-`is_scalar_tower`, but when it is only a `semiring` we need to build it from scratch.
-The instance diamond in `compatible_smul` doesn't matter because it's in `Prop`.
--/
```

Comment:
I think this comment maybe still applies? There's certainly still an instance diamond for `compatible_smul`, and I believe `tmul_smul` remains unusable without this instance. There's no longer a non-defeq diamond issue, but the problem that `semimodule ℤ _` isn't a strong enough condition remains.

Reply:
ok, I added it back.